### PR TITLE
fix: #423 

### PR DESCRIPTION
**Pull request by zenixls2**
limit startTime and endTime in query for /api/v3/myTrades endpoint  
may have the same issue in futures and margin endpoint, need further check  
@c9s if you have time, could you check for these two?

### CODE REVIEW

File: pkg/exchange/binance/exchange.go [hunk 1]
```diff
@@ -1190,30 +1191,41 @@ func (e *Exchange) QueryTrades(ctx context.Context, symbol string, options *type
...
+			endTime = options.EndTime.UnixNano() / int64(time.Millisecond)
+		}
+
+		for t = startTime; t < endTime; t += ONE_DAY_MILLISEC {
+			req.StartTime(t)
+			et := t + ONE_DAY_MILLISEC
+			if et > endTime {
+				et = endTime
+			}
+			req.EndTime(et)
```
Comment (c9s) [inline pkg/exchange/binance/exchange.go:1215]: If you used both startTime and endTime, you can not query the time range more than 24 hours.

**Comment by c9s** [comment]
I have fixed this in v1.27.0

**Comment by zenixls2** [comment]
> I have fixed this in v1.27.0

I don't think it's a perfect solution.  
Cli interface should provide same result no matter which exchange it connects to.  
Also the deduction of 1 millisecond seems not necessary

**Comment by c9s** [comment]
> > I have fixed this in v1.27.0
> 
> I don't think it's a perfect solution. Cli interface should provide same result no matter which exchange it connects to. Also the deduction of 1 millisecond seems not necessary

What do you want to fix?

**Comment by c9s** [comment]
You need to consider a situation -- when there is no trade between the given start time and end time, the sync will stop even if you have trade data in the later time range.

**Comment by c9s** [comment]
This is the reason that StartTime and EndTime can not be given at the same time.

File: pkg/exchange/binance/exchange.go [hunk 2]
```diff
@@ -33,6 +33,7 @@ const BinanceUSBaseURL = "https://api.binance.us"
 const BinanceUSWebSocketURL = "wss://stream.binance.us:9443"
 const WebSocketURL = "wss://stream.binance.com:9443"
 const FuturesWebSocketURL = "wss://fstream.binance.com"
+const ONE_DAY_MILLISEC = 86400000
```
Comment (c9s) [inline pkg/exchange/binance/exchange.go:36]: With the Go convention, for constant please use camel case~

**Comment by c9s** [comment]
add link #423

**Comment by c9s** [comment]
oh! I guess I fixed the QueryClosedOrders but I didn't fix the QueryTrade .....

**Comment by c9s** [comment]
https://binance-docs.github.io/apidocs/spot/en/#account-trade-list-user_data 

looks like the 24 hours limit only happens to the closed orders?

**Pull request closed without merging**